sky/sdk/lib/widgets/animated_container.dart

```diff
@@ -99,109 +211,39 @@ class AnimatedContainer extends AnimatedComponent {
-  ImplicitlyAnimatedValue<double> _height;
+  List<AnimationIntention> intentions;
+  dynamic tag; // Used by intentions to determine desired state.
 
   void initState() {
-    _updateFields();
-  }
-
-  void syncFields(AnimatedContainer source) {
-    child = source.child;
-    constraints = source.constraints;
-    decoration = source.decoration;
-    margin = source.margin;
-    padding = source.padding;
-    width = source.width;
-    height = source.height;
-    _updateFields();
-  }
-
-  void _updateFields() {
-    _updateConstraints();
-    _updateDecoration();
-    _updateMargin();
-    _updatePadding();
-    _updateTransform();
-    _updateWidth();
-    _updateHeight();
-  }
-
-  void _updateField(dynamic value, ImplicitlyAnimatedValue animatedValue, Function initField) {
-    if (animatedValue != null)
-      animatedValue.value = value;
-    else if (value != null)
-      initField();
-  }
-
-  void _updateConstraints() {
-    _updateField(constraints, _constraints, () {
-      _constraints = new ImplicitlyAnimatedValue<BoxConstraints>(new AnimatedBoxConstraintsValue(constraints), duration);
-      watch(_constraints.performance);
-    });
-  }
-
-  void _updateDecoration() {
-    _updateField(decoration, _decoration, () {
-      _decoration = new ImplicitlyAnimatedValue<BoxDecoration>(new AnimatedBoxDecorationValue(decoration), duration);
-      watch(_decoration.performance);
-    });
-  }
-
-  void _updateMargin() {
-    _updateField(margin, _margin, () {
-      _margin = new ImplicitlyAnimatedValue<EdgeDims>(new AnimatedEdgeDimsValue(margin), duration);
-      watch(_margin.performance);
-    });
-  }
-
-  void _updatePadding() {
-    _updateField(padding, _padding, () {
-      _padding = new ImplicitlyAnimatedValue<EdgeDims>(new AnimatedEdgeDimsValue(padding), duration);
-      watch(_padding.performance);
-    });
-  }
-
-  void _updateTransform() {
-    _updateField(transform, _transform, () {
-      _transform = new ImplicitlyAnimatedValue<Matrix4>(new AnimatedValue<Matrix4>(transform), duration);
-      watch(_transform.performance);
-    });
-  }
-
-  void _updateWidth() {
-    _updateField(width, _width, () {
-      _width = new ImplicitlyAnimatedValue<double>(new AnimatedValue<double>(width), duration);
-      watch(_width.performance);
-    });
-  }
-
-  void _updateHeight() {
-    _updateField(height, _height, () {
-      _height = new ImplicitlyAnimatedValue<double>( new AnimatedValue<double>(height), duration);
-      watch(_height.performance);
-    });
+    if (intentions == null) {
+      intentions = [
+        new ImplicitlySyncConstraintsIntention(duration),
+        new ImplicitlySyncDecorationIntention(duration),
+        new ImplicitlySyncMarginIntention(duration),
+        new ImplicitlySyncPaddingIntention(duration),
+        new ImplicitlySyncTransformIntention(duration),
+        new ImplicitlySyncWidthIntention(duration),
+        new ImplicitlySyncHeightIntention(duration),
+      ];
```

Comment:
Yeah I've moved this to a helper and had Material call the helper. This allowed me to drop the AnimationContainer's Duration as well.